src/de.rs

```diff
@@ -532,16 +551,67 @@ fn ensure_supported(value: &Value<'_>) -> Result<bool> {
     ))
 }
 
+/// A helper struct for deserializing enums containing unit variants.
+struct UnitEnumAccess {
+    variant: String,
+}
+
+impl<'de> de::EnumAccess<'de> for UnitEnumAccess {
+    type Error = Error;
+    type Variant = UnitOnlyVariant;
+
+    fn variant_seed<V>(self, seed: V) -> Result<(V::Value, Self::Variant)>
+    where
+        V: de::DeserializeSeed<'de>,
+    {
+        let id = StrDeserializer::<Error>::new(&self.variant);
+        let v = seed.deserialize(id)?;
```

Comment:
You should be able to do `let v = seed.deserialize(self.variant.into_deserializer())?;`